### PR TITLE
improve back end debug printing of intermediate code

### PR DESCRIPTION
**Pull request by WalterBright**
It's a lot easier to read when blocks are numbered rather than random pointer values.

### CODE REVIEW

**Review by stefan-koch-sociomantic — APPROVED** [review]
This is a step into the right direction.

File: src/ddmd/backend/debug.c [hunk 1]
```diff
@@ -318,25 +319,34 @@ void WRblock(block *b)
...
+            printf(" b_ret=B%d", b->BS.BI_FINALLY.b_ret ? b->BS.BI_FINALLY.b_ret->Bnumber : 0);
 #if MARS
         if (b->Bsrcpos.Sfilename)
             printf(" %s(%u)", b->Bsrcpos.Sfilename, b->Bsrcpos.Slinnum);
 #endif
         printf("\n");
-        if (b->Belem) elem_print(b->Belem);
+        if (b->Belem)
+        {
+            if (0)
```
Comment (JinShil) [inline src/ddmd/backend/debug.c]: Should this be `false` instead of `0`?

**Review by JinShil — CHANGES_REQUESTED** [review]
~I'm not able to comment on the functionality of this PR, as I don't quite understand what it does.  But I've reviewed the code itself, and have found no issues.~

Sorry a didn't mean to send that, and apparently Github won't allow me to delete it.

File: src/ddmd/backend/debug.c (hunk 1 above)
Comment (JinShil) [reply src/ddmd/backend/debug.c]: The reason I ask is because of #7310

**Review by JinShil — APPROVED** [review]

File: src/ddmd/backend/debug.c (hunk 1 above)
Comment (WalterBright) [reply src/ddmd/backend/debug.c]: `0` is never going to be an incorrect argument to `if`.

**Pull request merged**